### PR TITLE
Add support for CD-Guard

### PR DESCRIPTION
**Pull request by TheRogueArchivist**
* Add support for detecting CD-Guard, as well as notes.

* Add CD-Guard to README.

### CODE REVIEW

File: BurnOutSharp/ProtectionType/CDGuard.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            {
+                // Found in "cdguard.dll" in Redump entry 97142.
+                bool match = pex.ExportTable.ExportDirectoryTable.Name == "CDGUARD.DLL";
+                if (match)
+                    return "CD-Guard Copy Protection System";
+
+                // Found in "cdguard.dll" in IA item "pahgeby-he3hakomkou".
+                match = pex.ExportTable.ExportDirectoryTable.Name == "cdguard.dll";
+                if (match)
+                    return "CD-Guard Copy Protection System";
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/CDGuard.cs:46]: Why not combine and do a case-insensitive check? Or is there a worry that something like `CDGuard.DLL` would get caught unintentionally?

File: BurnOutSharp/ProtectionType/CDGuard.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            {
+                // Found in "Randevu.exe" in Redump entry 97142.
+                bool match = pex.ImportTable.ImportDirectoryTable.Any(idte => idte.Name == "CDGUARD.DLL");
+                if (match)
+                      return "CD-Guard Copy Protection System";
+
+                // Not found in any known sample, but presumably would be found in the game EXE for a game that uses the same driver version as IA item "pahgeby-he3hakomkou".
+                match = pex.ExportTable.ExportDirectoryTable.Name == "cdguard.dll";
+                if (match)
+                    return "CD-Guard Copy Protection System";
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/CDGuard.cs:60]: Same question here as above.

File: BurnOutSharp/ProtectionType/CDGuard.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string> files)
+        {
+            var matchers = new List<PathMatchSet>
+            {
+                // Found in Redump entry 97142.
+                new PathMatchSet(new PathMatch("cdguard.dll", useEndsWith: true), "CD-Guard Copy Protection System"),
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/CDGuard.cs:72]: Do you think it's necessary to put the directory separator in front of those? Or is it unlikely to get false positives?

File: BurnOutSharp/ProtectionType/CDGuard.cs (hunk 2 above)
Comment (TheRogueArchivist) [reply BurnOutSharp/ProtectionType/CDGuard.cs:60]: Ditto

File: BurnOutSharp/ProtectionType/CDGuard.cs (hunk 3 above)
Comment (TheRogueArchivist) [reply BurnOutSharp/ProtectionType/CDGuard.cs:72]: I'm not worried about that atm personally, but I can add it if you'd like.

**Review by mnadareski — APPROVED** [review]

**Pull request merged**